src/app/term/term-details/term-details.component.ts

```diff
@@ -16,248 +16,182 @@ limitations under the License.
-          this.editableForm.aliases.push(item);
-        });
-      }
-      if (this.mcTerm.semanticLinks) {
-        this.mcTerm.semanticLinks.forEach(link => {
-          if (link.linkType === 'New Version Of') {
-            this.compareToList.push(link.target);
-          }
-        });
-      }
-
-      if (this.mcTerm.semanticLinks) {
-        this.mcTerm.semanticLinks.forEach(link => {
-          if (link.linkType === 'Superseded By') {
-            this.compareToList.push(link.target);
-          }
-        });
-      }
-
-      if (this.mcTerm != null) {
-        this.hasResult = true;
-        this.watchDataModelObject();
-      }
-      this.title.setTitle(`Term - ${this.mcTerm?.label}`);
-    }
-    );
+  termDetails(): any {
+    const termId = this.uiRouterGlobals.params.id;
+    this.terminologyId = this.uiRouterGlobals.params.terminologyId;
+    this.resourcesService.term
+      .get(this.terminologyId, termId)
+      .subscribe((termDetailResult: TermDetailResponse) => {
+        this.mcTerm = termDetailResult.body;
+
+        this.resourcesService.terminology
+          .get(this.mcTerm.model)
+          .subscribe((result: TerminologyDetailResponse) => {
+            this.parentLabel = result.body.label;
+          });
+
+        if (this.mcTerm.semanticLinks) {
+          this.mcTerm.semanticLinks.forEach((link) => {
+            if (link.linkType === 'New Version Of') {
+              this.compareToList.push(link.target);
+            }
+          });
+        }
+
+        if (this.mcTerm.semanticLinks) {
+          this.mcTerm.semanticLinks.forEach((link) => {
+            if (link.linkType === 'Superseded By') {
+              this.compareToList.push(link.target);
+            }
+          });
+        }
+        this.access = this.securityHandler.elementAccess(this.mcTerm);
+        this.originalTerm = Object.assign({}, this.mcTerm);
+        this.title.setTitle(`Term - ${this.mcTerm?.label}`);
+      });
   }
 
-  onCancelEdit() {
-    this.errorMessage = '';
+  cancel() {
+    this.editingService.stop();
+    this.mcTerm = Object.assign({}, this.originalTerm);
     this.editMode = false; // Use Input editor whe adding a new folder.
   }
 
-  watchDataModelObject() {
-    const access: any = this.securityHandler.elementAccess(this.mcTerm);
-    if (access !== undefined) {
-      this.showEdit = access.showEdit;
-      this.showPermission = access.showPermission;
-      this.showDelete = access.showDelete;
-      this.showFinalise = access.showFinalise;
-      this.showNewVersion = access.showNewVersion;
+  save() {
+    if (this.validatorServive.validateLabel(this.mcTerm.definition)) {
+
+      this.resourcesService.term
+        .update(this.terminologyId, this.mcTerm.id, this.mcTerm)
+        .subscribe(
+          () => {
+            this.messageHandler.showSuccess('Term updated successfully.');
+            this.editingService.stop();
+            this.editMode = false;
+            this.errorMessage = '';
```

Comment:
I've noticed this `errorMessage` property in these detail views, but does it actually have a purpose anymore?